posts/2021-12-14-log4j2-cve.md

```diff
@@ -0,0 +1,58 @@
+---
+title: Grails Framework and Log4j 2 Vulnerability
+date: Dec 14, 2021
+description: Most applications are not affected since Grails uses Logback by default
+author: Sergio del Amo
+image: 2019-12-20.jpg
+---
+
+# [%title]
+
+[%author]
+
+[%date]
+
+
+Vulnerability [CVE-2021-44228](https://nvd.nist.gov/vuln/detail/CVE-2021-44228) may allow an attacker to remotely execute code. It affects [Log4j](https://logging.apache.org/log4j/2.x/)'s `log4j-core` jar. [Log4j v2.15.0](https://repo1.maven.org/maven2/org/apache/logging/log4j/log4j-core/2.15.0/) fixes it.
+
+## Are Grails applications vulnerable?
+
+Since Grails 3.0.0, Grails applications use [Logback](http://logback.qos.ch). Thus, **most Grails applications are not affected**.
+
+However, your application would be vulnerable if it includes untrusted input in the log messages and pulls `log4j-core` as a transitive dependency.
+
+## Are Grails 2 applications vulnerable?
+
+CVE-2021-44228 only affects log4j versions between 2.0 and 2.14.1. Grails 2 application used Log4j 1.x. However, you should consider migrating as soon as possible to the latest version of Grails. Grails 2 is EOL (End of Life).  
```

Comment:
Even though Grails 2 applications used Log4J 1, we should suggest that they verify that they are not pulling in a vulnerable version of Log4J

Reply:
addressed via https://github.com/grails/grails-static-website/pull/226/commits/327a82930316cc99089d35b0887b0be1d68e088f